### PR TITLE
 docs: suggestions for plugin API and CLI reference docs

### PR DESCRIPTION
**Pull request by kdmccormick**
Some tweaks I developed as I worked through the docs for the first time to build my first plugin. See my commit messages for more details.

Supersedes https://github.com/overhangio/tutor/pull/564

## Screenshots of changed sections

api.rst intro:

image
_________________________
api.rst commands:

image
________________________
gettingstarted.rst YAML plugins:

image
________________________
gettingstarted.rst Python plugins:
image

### CODE REVIEW

File: docs/plugins/api.rst [hunk 1]
```diff
@@ -1,7 +1,20 @@
 Plugin API
 ==========
 
-Plugins can affect the behaviour of Tutor at multiple levels. First, plugins can define new services with their Docker images, settings and the right initialisation commands. To do so you will have to define custom :ref:`config <plugin_config>`, :ref:`patches <plugin_patches>`, :ref:`hooks <plugin_hooks>` and :ref:`templates <plugin_templates>`. Then, plugins can also extend the CLI by defining their own :ref:`commands <plugin_command>`.
+A Tutor plugin is defined either by a YAML file or a Python package.
```
Comment (regisb) [inline docs/plugins/api.rst:4]: I feel like this sentence should be placed after the more general description of what plugins can achieve. 

I propose the following:

```
Plugins can affect the behaviour of Tutor at multiple levels:

* Add new settings or modify existing one in the Tutor configuration (see :ref:`config <plugin_config>`).
* Add new templates to the Tutor project environment or modify existing ones (see :ref:`patches <plugin_patches>`, :ref:`templates <plugin_templates>` and :ref:`hooks <plugin_hooks>`).
* Add custom commands to the Tutor CLI (see :ref:`command <plugin_command>`).

There exists two different APIs to create Tutor plugins: either with YAML files or Python packages. YAML files are more simple to create, but are limited to just configuration and template patches.
```

Note that you should not to line-wrap the docs. (and below as well)

File: docs/plugins/api.rst [hunk 2]
```diff
@@ -207,11 +223,21 @@ You can even define subcommands by creating `command groups <https://click.palle
     def command():
         pass
 
-    @click.command(help="I'm a plugin subcommand")
+    @command.command(help="I'm a plugin subcommand")
```
Comment (regisb) [inline docs/plugins/api.rst:226]: Wow, I had no idea that this was possible! What I usually do is that I add `command.add_command` statements, like here: https://github.com/overhangio/tutor-xqueue/blob/fb9e666814c1bdf6d00f8b80f94508b0404d5315/tutorxqueue/plugin.py#L201
But now I forgot to add this `add_command` statement to the docs, so you solution is better.

File: docs/plugins/gettingstarted.rst [hunk 3]
```diff
@@ -14,7 +14,9 @@ YAML files that are stored in the tutor plugins root folder will be automaticall
 
 On Linux, this points to ``~/.local/share/tutor-plugins``. The location of the plugin root folder can be modified by setting the ``TUTOR_PLUGINS_ROOT`` environment variable.
 
-YAML plugins need to define two extra keys: "name" and "version". Custom CLI commands are not supported by YAML plugins.
+YAML plugins must define two special top-level keys: ``name`` and ``version``.
+Then, YAML plugins may use four top-level keys to customize Tutor's behavior: ``config``, ``patches``, ``hooks``, and ``templates``.
```
Comment (regisb) [inline docs/plugins/gettingstarted.rst:18]: Unfortunately, hooks and templates are not supported either in yaml files :-/ See: https://github.com/overhangio/tutor/issues/499

**Review by regisb — CHANGES_REQUESTED** [review]

File: docs/plugins/api.rst (hunk 1 above)
Comment (kdmccormick) [reply docs/plugins/api.rst]: That it much more streamlined, thanks.

**Comment by kdmccormick** [comment]
Ready for another round.

**Review by regisb — APPROVED** [review]

**Comment by regisb** [comment]
These changes are great, thanks a lot.

**Pull request merged**